packages/headless/src/features/generated-answer/generated-answer-request.ts

```diff
@@ -25,11 +29,43 @@ import type {
   SearchSection,
 } from '../../state/state-sections.js';
 import {getFacets} from '../../utils/facet-utils.js';
+import type {ContextPayload} from '../context/context-state.js';
+import type {AnyFacetRequest} from '../facets/generic/interfaces/generic-facet-request.js';
 
 type StateNeededByGeneratedAnswerStream = ConfigurationSection &
   SearchSection &
   GeneratedAnswerSection;
 
+export interface AnswerApiQueryParams {
```

Comment:
Please sync with @gleblanc-coveo as I want to make sure this type fits the goal they currently have to match the payload of the Search API.

It turns out that it might actually be 100% the same payload? Not even a Partial?

Reply:
You could get inspired by the following: https://github.com/coveo/ui-kit/blob/51f6eca0eab3428fb946fd59d4b07b1996541b52/packages/headless/src/api/search/recommendation/recommendation-request.ts#L23

Reply:
where exactly do you see the `SearchApiQueryParams` ?

<img width="286" height="381" alt="image" src="https://github.com/user-attachments/assets/3de85d39-00ac-4bf5-8c7b-7e1922c847e7" />

Also on your note to build it using other parts, I kinda like having the type declared there as its really easy to see whats in it versus having to do mental gymnastics but that could just be me.